File: ShoppingV3Assets/app/src/main/java/dk/itu/shoppingv3assets/ShoppingActivity.java
package dk.itu.shoppingv3assets;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class ShoppingActivity extends AppCompatActivity {
    //Model: Database of items
    private static ItemsDB itemsDB;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.shopping);

        ItemsDB.setContext(ShoppingActivity.this);
        itemsDB = ItemsDB.get();

        //Text Fields
        TextView newWhat = findViewById(R.id.what_text);
        TextView newWhere = findViewById(R.id.where_text);

        Button listItems = findViewById(R.id.list);
        listItems.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(ShoppingActivity.this, ListActivity.class);
                startActivity(intent);
            }
        });

        Button addItem = findViewById(R.id.add_button);
        // adding a new thing
        addItem.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String whatS = newWhat.getText().toString().trim();
                String whereS = newWhere.getText().toString().trim();
                if ((whatS.length() > 0) && (whereS.length() > 0)) {
                    itemsDB.addItem(whatS, whereS);
                    newWhat.setText("");
                    newWhere.setText("");
                } else
                    Toast.makeText(ShoppingActivity.this, R.string.empty_toast, Toast.LENGTH_LONG).show();
            }
        });
    }
}

File: ShoppingV3Kotlin/app/src/main/java/dk/itu/myshoppingv3kotlin/ShoppingActivity.kt
package dk.itu.myshoppingv3kotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast

class ShoppingActivity : AppCompatActivity() {
    // Model: Database of items
    private lateinit var itemsDB: ItemsDB

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.shopping)

        itemsDB = ItemsDB.get()

        //Text Fields
        val newWhat: TextView = findViewById(R.id.what_text)
        val newWhere: TextView = findViewById(R.id.where_text)

        val listItems: Button = findViewById(R.id.list)
        listItems.setOnClickListener {
            val intent = Intent(this, ListActivity::class.java)
            startActivity(intent)
        }

        val addItem: Button = findViewById(R.id.add_button)
        addItem.setOnClickListener {
            val whatS = newWhat.text.toString().trim()
            val whereS = newWhere.text.toString().trim()
            if ((whatS.length > 0) && (whereS.length > 0)) {
                itemsDB.addItem(whatS, whereS)
                newWhat.text = ""
                newWhere.text = ""
            } else {
                Toast.makeText(this, R.string.empty_toast, Toast.LENGTH_LONG).show();
            }
        }
    }
}